tests/system/Router/RouterTest.php

```diff
@@ -925,4 +931,34 @@ public static function provideRedirectRoute(): iterable
             ['post/(:segment)/comment/(:any)', 'article/(:num)/(:any)', 'alias', '/article/1/2', 'post/1/comment/2', 'alias'],
         ];
     }
+
+    public function testRoutePlaceholderAnyWithMultipleSegmentsParamFalse(): void
+    {
+        $this->createRouteCollection();
+        $this->collection->get('product/(:any)', 'Catalog::productLookup/$1');
+        $router = new Router($this->collection, $this->request);
+
+        $router->handle('product/123/456');
+
+        $this->assertSame('\Catalog', $router->controllerName());
+        $this->assertSame('productLookup', $router->methodName());
+        $this->assertSame(['123', '456'], $router->params());
+    }
+
+    public function testRoutePlaceholderAnyWithMultipleSegmentsParamTrue(): void
+    {
+        $routingConfig                           = new Routing();
+        $routingConfig->multipleSegmentsOneParam = true;
+        Factories::injectMock('config', Routing::class, $routingConfig);
+
+        $this->createRouteCollection($routingConfig);
+        $this->collection->get('product/(:any)', 'Catalog::productLookup/$1');
+        $router = new Router($this->collection, $this->request);
+
+        $router->handle('product/123/456');
+
+        $this->assertSame('\Catalog', $router->controllerName());
+        $this->assertSame('productLookup', $router->methodName());
+        $this->assertSame(['123/456'], $router->params());
+    }
```

Comment:
Would you please add a test showing that queries are not included?
```
product/123/456?id=789
```

Reply:
@MGatner Do you mean `$router->handle('product/123/456?id=789');`?
If so, it is not needed, because the parameter is `$uri`. It never contains queries.
In other words, if `product/123/456?id=789` is passed to `handle()`, the client code is wrong.

https://github.com/codeigniter4/CodeIgniter4/blob/b28af1dc0609caec4b4017379361c073594d8b91/system/CodeIgniter.php#L826-L830